client.go

```diff
@@ -223,23 +234,11 @@ func (c Client) HardwareAddr() net.HardwareAddr {
 
 // firstIPv4Addr attempts to retrieve the first detected IPv4 address from an
 // input slice of network addresses.
-func firstIPv4Addr(addrs []net.Addr) (net.IP, error) {
+func firstIPv4Addr(addrs []netip.Addr) (netip.Addr, error) {
 	for _, a := range addrs {
-		if a.Network() != "ip+net" {
-			continue
-		}
-
-		ip, _, err := net.ParseCIDR(a.String())
-		if err != nil {
-			return nil, err
-		}
-
-		// "If ip is not an IPv4 address, To4 returns nil."
-		// Reference: http://golang.org/pkg/net/#IP.To4
-		if ip4 := ip.To4(); ip4 != nil {
-			return ip4, nil
+		if a.Is4() {
+			return a, nil
 		}
 	}
-
-	return nil, nil
+	return netip.Addr{}, nil
```

Comment:
I made a logic mistake here: this should return an error if no IPv4 address exists on the interface.